### ISSUE TITLE
giou loss and object loss are opposite

### ISSUE THREAD

**Issue by wojiaohumaocheng** [ultralytics/yolov5#1136]
what is the problem about these losses?

![image](https://user-images.githubusercontent.com/25666696/95968776-734c2d00-0e40-11eb-8678-35f59bf16f68.png)

**Comment by Aktcob** [comment]
train little part of ur datasets. make sure that no annotation error in ur datasets.

**Comment by abhiagwl4262** [comment]
@glenn-jocher I have few doubts that occured in my mind. 

lbox += (1.0 - giou).mean() -> THis loss will try to get higher GIOU or better localisation. 
tobj[b, a, gj, gi] = (1.0 - model.gr) + model.gr * giou.detach().clamp(0).type(tobj.dtype) -> THis loss tries to make confidence close to GIOU because if bounding box is not accurate then confidence should also be lower. 

the GIOU loss is calculated only for matched locations while obj_loss is calculated for all the locations which are `20*20*3` + `40*40*3 `+ `80*80*3.` 

Still the weightage of GIOU loss is very less than objectness loss. Do you have any explanation for the same?

**Comment by glenn-jocher** [comment]
@abhiagwl4262 yes your interpretation of the losses is correct. We are taking the mean loss over all of the individual elements though, so if box has 100 elements and obj has 10000 elements, we are averaging their losses before summing:

loss_p3 = mean(box_p3) + mean(obj_p3)

 The actual loss balancing hyperparameters were evolved on COCO, and lower box gain found to improve performance WRT obj/cls loss gains. Of course different datasets may benefit from different hyperparameters too, as well as different tasks (finetuning vs training from scratch).
https://github.com/ultralytics/yolov5/blob/747c2653eecfb870b1ed40b1e00e0ef209b036e9/data/hyp.scratch.yaml#L13-L17

**Comment by abhiagwl4262** [comment]
@glenn-jocher 

1. That means the mean value of lbox must be coming very high than mean value of lobj. What can be reason of that ?
2. You evolved on COCO. Do you mean, you ran 300 generations of evolution on COCO? That might have taken months ?
